zips/zip-0312.rst

```diff
@@ -212,11 +175,11 @@ involves these steps:
 Round Two - Signature Share Generation
 ''''''''''''''''''''''''''''''''''''''
 
-In Round Two, the Coordinator generates a random scalar ``randomizer`` by calling
-``randomizer_generate`` and sends it to each signer, over a confidential and
-authenticated channel, along with the message and the set of signing
-commitments. (Note that this differs from regular FROST which just requires an
-authenticated channel.)
+In Round Two, the Coordinator picks a message and sends it to each signer along
+with the commitments received in Round One, over an authenticated and encrypted
+channel. In Re-Randomized FROST, an additional value called `randomizer` needs
+to be generated by the Coordinator and sent along with the message and
+commitments.
```

Comment:
Does the math not work out if each participant generates a share of alpha and commits to it in round 1, then uses their share in round 2? If it didn't then I guess the way to remove the trust assumption on the Coordinator here would be to still do that commitment in round 1, and then have a round 1.5 that is just participants revealing their alpha shares, but that would turn this into a three-round protocol, or two-round with preprocessing (where participants precompute and save their shares of both R and alpha), which isn't ideal for some usage scenarios where the Coordinator _is_ trusted. Maybe that should be left as an option?

Reply:
One way to reduce the requirements on the Coordinator is to have it generate the randomizer by hashing the message, commitments, and an extra seed. The participants then recalculate it using that hash. The Coordinator can grind on the seed in order to create a subliminal channel, but this means you're not trusting the Coordinator's RNG. (Also, no participant can grind on their contribution in order to influence the randomizer, because the Coordinator only reveals the seed after it has the message and commitments.)

Reply:
I tried @daira 's approach in 62833e6c